src/components/queryBuilder/views/LogsQueryBuilder.tsx

```diff
@@ -0,0 +1,138 @@
+import React, { useEffect, useState } from 'react';
+import { ColumnsEditor } from '../ColumnsEditor';
+import { BuilderMode, Filter, TableColumn, OrderBy, QueryBuilderOptions, SelectedColumn, ColumnHint } from 'types/queryBuilder';
+import { ColumnSelect } from '../ColumnSelect';
+import { Switch } from '../Switch';
+import { OtelVersionSelect } from '../OtelVersionSelect';
+import { OrderByEditor } from '../OrderByEditor';
+import { LimitEditor } from '../LimitEditor';
+import { FiltersEditor } from '../FilterEditor';
+import allLabels from 'labels';
+import { getColumnByHint } from 'components/queryBuilder/utils';
+
+interface LogsQueryBuilderProps {
+  allColumns: TableColumn[];
+  builderOptions: QueryBuilderOptions,
+  onBuilderOptionsChange: (builderOptions: QueryBuilderOptions) => void;
+}
+
+export const LogsQueryBuilder = (props: LogsQueryBuilderProps) => {
+  const { allColumns, builderOptions, onBuilderOptionsChange } = props;
+  const [otelEnabled, setOtelEnabled] = useState<boolean>(false);
+  const [otelVersion, setOtelVersion] = useState<string>('');
+  const [selectedColumns, setSelectedColumns] = useState<SelectedColumn[]>([]);
+  const [timeColumn, setTimeColumn] = useState<SelectedColumn>();
+  const [logLevelColumn, setLogLevelColumn] = useState<SelectedColumn>();
+  const [messageColumn, setMessageColumn] = useState<SelectedColumn>();
+  const [liveView, setLiveView] = useState<boolean>(false);
+  const [orderBy, setOrderBy] = useState<OrderBy[]>([]);
+  const [limit, setLimit] = useState<number>(100);
+  const [filters, setFilters] = useState<Filter[]>([]);
+  const labels = allLabels.components.LogsQueryBuilder;
+
+  useEffect(() => {
+    if (!builderOptions) {
+      return;
+    }
+
+    builderOptions.meta?.otelEnabled !== undefined && setOtelEnabled(builderOptions.meta.otelEnabled);
+    builderOptions.meta?.otelVersion && setOtelVersion(builderOptions.meta.otelVersion);
+    setTimeColumn(getColumnByHint(builderOptions, ColumnHint.Time));
+    setLogLevelColumn(getColumnByHint(builderOptions, ColumnHint.LogLevel));
+    setMessageColumn(getColumnByHint(builderOptions, ColumnHint.LogMessage));
+    builderOptions.columns && setSelectedColumns(builderOptions.columns.filter(c => c.hint === undefined));
```

Comment:
For self-education: how us `hint` related to selected columns?

Reply:
I added the `hint` field to the `SelectedColumn` type so that the UI and query generator will have more information on the purpose of the column.

### In the UI
In the UI logic, we can identify which dropdown a column should go in to. An interesting example is selecting the time column in the Time Series view. Both Time Series and Logs view contain a dropdown for selecting a time column, so if I select the time column in one of these views, it will automatically populate the correct dropdown when I switch views. This is because `builderOptions.columns` contains a `SelectedColumn` with `hint: ColumnHint.Time` in it. Even if I switch to a view without a dedicated time dropdown (such as TableView), this information is retained and used when switching back to Time Series view.

`{ columns: [{ name: 'my_timestamp' type: 'datetime', hint: ColumnHint.Time }] }`

In the snippet of code you commented on here, the goal is to only show columns that don't have their own dropdown.
This is also used when converting back from raw SQL to the query builder. If the SQL parser determines a column to be a `time` field, it can re-add these hints so that the Query Builder will be pre-loaded with the correct values. Instead of going into the column selector, it will go into the time field dropdown.

### In the SQL generator

The SQL generator code also makes use of the column hint. As we add more advanced features (such as logs/traces), we need a way to reference columns without adding too many custom fields to the `QueryBuilderOptions` type.

I added a helper function for fetching these columns:
```ts
const timeColumn = getColumnByHint(query.builderOptions, ColumnHint.Time);
```
There aren't any scenarios where multiple columns will have the same hint, so we can reliably identify the time column by using this (assuming it was selected from a time field dropdown in the UI).

Previously to get a specialized field, the code was using a dedicated field on `builderOptions` (such as `builderOptions.timeField` and `builderOptions.timeFieldType`). The use of `SelectedColumn` provides helpful context so that we can do things such as:
- adding column aliases
- applying type conversion functions
- adding column hints (so we can apply functions/aliases to special columns for logs/trace views)